File: nodes/throw-error/throw-error.process.js
export default async ({inputs, settings, config}) => {

  const message = inputs.message || "No error message provided";
  const context = inputs.context || "";
  const errorType = inputs.errorType || "CustomError";

  // Construct the full error message
  let fullMessage = message;
  if (context) {
    fullMessage += `\n\nContext: ${context}`;
  }

  // Create a custom error with the specified type
  const error = new Error(fullMessage);
  error.name = errorType;
  
  // Add context as a property for debugging
  if (context) {
    error.context = context;
  }

  // Throw the error - this will be caught by the engine
  throw error;
}; 

File: nodes/throw-error/throw-error.process.py
async def process(inputs, settings, config, nodeConfig):
    """
    Process function for the Throw Error node.
    Raises an exception with a custom message and optional context.
    """
    message = inputs.get("message", "No error message provided")
    context = inputs.get("context", "")
    error_type = inputs.get("errorType", "CustomError")

    # Construct the full error message
    full_message = message
    if context:
        full_message += f"\n\nContext: {context}"

    # Create a custom exception class if needed
    if error_type != "CustomError":
        class CustomException(Exception):
            def __init__(self, message, context=None):
                super().__init__(message)
                self.context = context
                self.name = error_type
        
        # Raise the custom exception
        raise CustomException(full_message, context)
    else:
        # Use a standard exception with context
        exception = Exception(full_message)
        exception.context = context
        exception.name = error_type
        raise exception 